pkg/arvo/gen/story-read.hoon

```diff
@@ -0,0 +1,122 @@
+  tang+[leaf+"Error: desk {<desk>} does not exist." ~]
+=/  pax                   /(scot %p our)/[desk]/(scot cas)/story
+?:  !.^(? %cu pax)        tang+['Error: No story file found. Please use |story-init to create one.' ~]
+=/  tale                  .^(story %cx pax)
+=/  current-tako          .^(tako:clay %cs /(scot %p our)/[desk]/(scot cas)/tako/~)
+=/  current-yaki          .^(yaki:clay %cs /(scot %p our)/[desk]/(scot cas)/yaki/(scot %uv current-tako))
+:-  %tang
+(story-read [our desk cas] current-yaki tale)
+::::
+::  Remarks:
+::
+::  There are two recursions in the logging process:
+::  1. the outer loop, `commit-loop` which threads downwards into each commit by ancestor
+::  2. the inner loop, `ancestor-loop`, which threads left-to-right on reverse-ancestors
+::
+::  +story-read outputs a tang with the least-recent commits at the head of the list,
+::  even though we want most-recent commits to end up printing first.
+::  But because dojo prints tangs in reverse, we don't flop the results 
+::::
+++  story-read
+  |=  [[our=ship syd=^desk cas=case] current-commit=yaki:clay tale=story]
+  ^-  tang
+  %-  head  :: result from state
+  =|  state=[result=tang mergebase=(unit tako:clay)]
+  |-
+  ^-  _state
+  =*  commit-loop  $
+  =/  reverse-ancestors  (flop p.current-commit)
+  |-
+  =*  ancestor-loop  $
+  ?-    reverse-ancestors
+      ~
+    ::  stop here and return the current message
+    =/  msg=(list cord)  (msg-from-commit current-commit tale)
+    [(weld msg result.state) mergebase=~]
+  ::
+      [tako:clay ~]
+    =/  parent=tako:clay  i.reverse-ancestors
+    =/  parent-commit     .^(yaki:clay %cs /(scot %p our)/[syd]/(scot cas)/yaki/(scot %uv parent))
+    =/  msg               (msg-from-commit current-commit tale) 
+    ::  If there is a mergebase and we are visting it right now:
+    ::    stop here and clear the mergebase. 
+    ::    skip adding the mergebase's msg itself because it will be added through the other branch
+    ::  Otherwise, record the current message if exists
+    ?:  ?&(?=(^ mergebase.state) =(u.mergebase.state r.current-commit))
+      [result=result.state mergebase=~]
+    commit-loop(current-commit parent-commit, result.state (weld msg result.state))
+  ::
+      [tako:clay tako:clay ~]
+    ::
+    ::  mainline: ultimate base chain
+    ::  nowline: relative mainline
+    ::  sideline: side-chain, featurebranch
+    ::
+    ::  From the context of e, commit c is on its relative mainline, or nowline, while commit d is on its sideline
+    ::  %base  a--b-------------X  :: mainline
+    ::  %new      \--c------e--/   :: nowline
+    ::  %new2        \--d--/       :: sideline 
+    ::
+    ::  
+    =/  sideline=tako:clay               i.reverse-ancestors
+    =/  mainline=tako:clay               i.t.reverse-ancestors
+    =/  mergebases                       .^((list tako:clay) %cs /(scot %p our)/[syd]/(scot cas)/base-tako/(scot %uv mainline)/(scot %uv sideline))  :: XX base-tako ignores beak
```

Comment:
We have a strict 80 columns rule, which applies in a few files here.

"80 columns is not just a good idea, it's the law!"

Reply:
done in [`341e27b` (#5612)](https://github.com/urbit/urbit/pull/5612/commits/341e27b5c36ce8f6b20c138e606df84963891798)